src/main/java/util/xcmailr/XcMailrApi.java

```diff
@@ -25,45 +25,103 @@
      */
     public static XcMailrConfiguration getConfiguration()
     {
         // the property needs to be a valid URI in order to satisfy the Owner framework
         if (null == ConfigFactory.getProperty(TEMPORARY_CONFIG_FILE_PROPERTY_NAME))
         {
-            ConfigFactory.setProperty(TEMPORARY_CONFIG_FILE_PROPERTY_NAME, "file:this/path/should/never/exist/noOneShouldCreateMe.properties");
+            ConfigFactory.setProperty(TEMPORARY_CONFIG_FILE_PROPERTY_NAME,
+                                      "file:this/path/should/never/exist/noOneShouldCreateMe.properties");
         }
         return CONFIGURATION.computeIfAbsent(Thread.currentThread(), key -> {
             return ConfigFactory.create(XcMailrConfiguration.class);
         });
     }
 
+    /**
+     * Reset configuration properties for thread. Needed for unit tests to enable reset of configurations for individual
+     * test methods, which belong to one test class
+     */
+    protected static void resetConfigurationsForThread()
+    {
+        CONFIGURATION.put(Thread.currentThread(), ConfigFactory.create(XcMailrConfiguration.class));
+    }
+
+    /**
+     * Create temporary email with validity time stated in xcmailr.properties under xcmailr.temporaryMailValidMinutes
+     * option
+     * 
+     * @param email
+     *            full temporary mail's name with domain </br>
+     *            mind, that not all domains are acceptable acceptable domains : mailsink.xceptance.de, varmail.net,
+     *            varmail.co.uk, varmailservice.com, var-mail.com, mail.varmail.net, varmail.de, varmail.online,
+     *            varmail.international
+     */
     public static void createTemporaryEmail(String email)
     {
-        String url = getConfiguration().url() + "/create/temporaryMail/" + getConfiguration().apiToken() + "/" + email + "/"
-                     + getConfiguration().temporaryMailValidMinutes();
-        HttpUrl.Builder builder = HttpUrl.parse(url).newBuilder();
-        Response response = callXcMailr(builder.build());
+        final String url = getConfiguration().url() + "/create/temporaryMail/" + getConfiguration().apiToken() + "/" + email + "/" +
+                           getConfiguration().temporaryMailValidMinutes();
+        final HttpUrl.Builder builder = HttpUrl.parse(url).newBuilder();
+        final Response response = callXcMailr(builder.build());
 
         Assert.assertNotNull("XcMailr not reachable", response);
         Assert.assertEquals("Temporary Email could not be created", 200, response.code());
         LOGGER.debug("Email created: \"" + email + "\"");
         response.close();
     }
 
+    /**
+     * Get last received email with specified subject
+     * 
+     * @param email
+     *            email address on which expected email should arrive
+     * @param subject
+     *            regular expression to find in the emails subject
+     * @return json object of received message
```

Comment:
```suggestion
     * @return a String containing a JSON Object of the received message
```

However, if I take a look at the test, it seems as if we first get a JSON Array which then contains a JSON Object which contains our expected fields. Maybe we should describe this in our API Doc, too?